e2e_test/ddl/duplicate_column_name.slt

```diff
@@ -0,0 +1,73 @@
+# Test for duplicate column names in SELECT statements
+# Issue: https://github.com/risingwavelabs/risingwave/issues/24095
+
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+# Test 1: Duplicate column names in CTE - exact same case
+statement error
+with cte as (
+  select
+    1 as col,
+    2 as col
+)
+select cte.col from cte;
+
+# Test 2: Duplicate column names in simple SELECT
+statement error
+select 1 as col, 2 as col;
```

Comment:
duplicate column names in simple select is allowed in Postgres. But it is only not allowed when we want to refer to one of them.